### PR TITLE
Added rst files for datasets

### PR DESCRIPTION
**Pull request by SaitejaUtpala**
Generated .rst files for datasets module and added reference to  it in api-reference.rst

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -134,3 +134,6 @@ dmypy.json
 # Miscellaneous
 out.mp4
 examples/out.mp4
+
+#python virutal environment
```
Comment (ninamiolane) [inline .gitignore:138]: NIT:
```
# Python virtual environment
```

(Correct the typo and add a space after the `#`)

File: docs/datasets.rst [hunk 2]
```diff
@@ -0,0 +1,37 @@
+datasets package
```
Comment (ninamiolane) [inline docs/datasets.rst:1]: Very nice! Was this `.rst` automatically generated or did you write it?

If you wrote it, let's try to use the same syntax/headings than `geometry.rst` [here](https://raw.githubusercontent.com/geomstats/geomstats/master/docs/geometry.rst).

For example, writing:
```
********
Datasets
********
```

**Review by ninamiolane — APPROVED** [review]
Very nice! 🎉 Thank you for doing this. Did you generate the .rst automatically or did you write it?

File: docs/datasets.rst (hunk 2 above)
Comment (SaitejaUtpala) [reply docs/datasets.rst:1]: I generated it by `sphinx-apidoc` command.

**Pull request merged**